catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/PipelineRepository.java

```diff
@@ -311,8 +311,8 @@ private void updateTasks(Pipeline original, Pipeline updated) throws JsonProcess
     }
 
     private void updateTaskDescription(Task origTask, Task updatedTask) throws JsonProcessingException {
-      if (operation.isPut() && !nullOrEmpty(origTask.getDescription())) {
-        // Update description only when stored is empty to retain user authored descriptions
+      if (operation.isPut() && !nullOrEmpty(origTask.getDescription()) && updatedByBot()) {
```

Comment:
@sureshms needs a flag here to gate the updates even by bot.
Usecase would be 
1. Ingestion runs, ingests description from sources
2. User updates, next run of ingestion shouldn't be updating the descriptions.  Current behavior, we should continue to support this

Use case 2:
1. Ingestion runs, updates the description from source, and the admin enables a flag on the ingestion side, "allowOverridingDescriptions"
2. If the above flag is set if the update is sent by bot then we will override any changes from users side using UI

Reply:
@harshach, let's keep this PR simple.

In order to add overwrite from ingestion, we need to consider a few more things:
1. What fields can be overwritten - description, tags, what else? Currently, we aggregate tags in PUT requests to retain the tags. Second, overwriting tags, say from DBT can be problematic for governance workflows. 
2. If overwriting is set, then we should prevent users from updating them in OpenMetadata to avoid them updating the description, only to see it being lost due to overwriting.

Let's create a separate issue for this.